Source/WebKit/NetworkProcess/NetworkProcess.cpp

```diff
@@ -395,6 +395,9 @@ void NetworkProcess::createNetworkConnectionToWebProcess(ProcessIdentifier ident
 
 void NetworkProcess::addAllowedFirstPartyForCookies(WebCore::ProcessIdentifier processIdentifier, WebCore::RegistrableDomain&& firstPartyForCookies, LoadedWebArchive loadedWebArchive, CompletionHandler<void()>&& completionHandler)
 {
+    if (!HashSet<WebCore::RegistrableDomain>::isValidValue(firstPartyForCookies))
```

Comment:
Add this to NetworkProcess::webProcessWillLoadWebArchive too

Reply:
decltype(m_allowedFirstPartiesForCookies)

Reply:
It is a little more complicated than this, here is the type:
`HashMap<WebCore::ProcessIdentifier, std::pair<LoadedWebArchive, HashSet<WebCore::RegistrableDomain>>>`

I only care about the inner HashSet.

Reply:
> webProcessWillLoadWebArchive

webProcessWillLoadWebArchive doesn't modify the HashSet containing RegistrableDomains, only the top-level HashMap.